### ISSUE TITLE
Compilation error on Linux Centos7

### ISSUE THREAD

**Issue by angelsaj** [opesci/devito#659]
Dear Devito programmers,

I am compiling devito on a Linux Centos 7 distribution. I am using Miniconda3 and tcsh as shell. I followed the README.md instructions up to

```
git clone https://github.com/opesci/devito.git
cd devito
```

Then, since I have tcsh shell I used:

`source /home/user/activate.tcsh /path/to/miniconda3 `
instead of 
`source activate devito`
in which activate.tcsh is a file taken from [here](https://github.com/andywhite37/virtualenv.tcsh/blob/master/activate.tcsh) 

It worked fine. The problem arises when I type:

`pip install -e .`

I get the following error:
```
    /usr/bin/ld: /usr/local/lib64/libblas.a(zher2k.o): relocation R_X86_64_32 against `.rodata' can not be used when making a shared object; recompile with -fPIC
    /usr/bin/ld: /usr/local/lib64/libblas.a(scopy.o): relocation R_X86_64_PC32 against symbol `memcpy@@GLIBC_2.14' can not be used when making a shared object; recompile with -fPIC
    /usr/bin/ld: final link failed: Bad value
    collect2: error: ld returned 1 exit status
    error: Command "/usr/bin/gfortran -Wall -g -Wall -g -shared build/temp.linux-x86_64-3.7/numpy/linalg/lapack_litemodule.o build/temp.linux-x86_64-3.7/numpy/linalg/lapack_lite/python_xerbla.o -L/usr/local/lib64 -L/usr/lib/gcc/x86_64-redhat-linux/4.8.5 -L/usr/lib/gcc/x86_64-redhat-linux/4.8.5 -Lbuild/temp.linux-x86_64-3.7 -llapack -llapack -lblas -lblas -lgfortran -o build/lib.linux-x86_64-3.7/numpy/linalg/lapack_lite.cpython-37m-x86_64-linux-gnu.so" failed with exit status 1
    
    ----------------------------------------
Command "/home/angelo/miniconda3/bin/python -u -c "import setuptools, tokenize;__file__='/tmp/pip-install-k64g44le/numpy/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-record-u81zfgaw/install-record.txt --single-version-externally-managed --compile" failed with error code 1 in /tmp/pip-install-k64g44le/numpy/
You are using pip version 10.0.1, however version 18.1 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
```
I am quite unexperienced with this kind of problems, and I have no clue on what to do. Do you have any suggestions?

Please find attached a file containing the complete terminal output with the error message. Let me know if you need further info.

Thank you in advance,

Angelo

[screen_output.pdf](https://github.com/opesci/devito/files/2469446/screen_output.pdf)

**Comment by FabioLuporini** [comment]
Hi Angelo,

what gcc collection are you using?

**Comment by angelsaj** [comment]
Hi Fabio,

The gcc version is 4.8.5 20150623 (Redhat 4.8.5 - 28)


Angelo


Il ven 12 ott 2018, 09:08 Angelo Sajeva <angelo.saje@gmail.com> ha scritto:

**Comment by FabioLuporini** [comment]
yes, that's a bit old. Can you upgrade to anything more recent?

**Comment by angelsaj** [comment]
Thank you for the reply, I cannot temporarily access the cluster's node.
When I will, I will try to upgrade as you suggested.

I will keep you updated as soon as I have news.

Thank you again,

Angelo

**Comment by angelsaj** [comment]
I solved in the following way. I moved to bash. Followed the procedure in
README.md. It worked. Evidently there were problems caused by using tcsh. I
tested tutorials 01 in seismic modelling and 02 in RTM and they work. I
haven't installed new gcc as you suggested, but I have noticed that devito
now uses  Python 3.6.6 |Anaconda, Inc.| (default, Oct  9 2018, 12:34:16) [GCC
7.3.0]. So a new gcc was already there, it was just not linked correctly.


Thank you for the support,

Angelo
